File: meetupCpp/ownership.cpp
#include <iostream>

using namespace std;

void great(string name){
    cout<<"Hello, "<<name <<"!"<<endl;
}

void great_ref(string& name){
    cout<<"Hello, "<<name <<"!"<<endl;
}

int main() {
    string alice = "Alice";

    great_ref(alice);   //Borrow
    great(alice);       //Copy
    great(move(alice)); //Move

    great(alice);       //Use after move. Why not?
}

File: meetupRust/src/ownership.rs
fn great(name : String){
    println!("Hello, {}", name);
}

fn great_ref(name : &String){
    println!("Hello, {}", name);
}

fn main() {
    let alice = "Alice".to_string();

    great_ref(&alice); //Borrow
    great(alice.clone()); //Copy
    great(alice); //Move

    // great(alice); // error[E0382]: use of moved value: `alice`

}